### PR TITLE
Implements std.meta.ArgsTuple.

### PR DESCRIPTION
**Pull request by MasterQ32**
Implements a helper function that is useful for creating generic function calls: `std.meta.ArgsTuple` calculates a tuple type for any given non-generic, non-variadic function type that can be passed into `@call()` to invoke the function.

### CODE REVIEW

**Comment by Rocknest** [comment]
It seems that the code is almost identical to #6416

**Comment by alexnask** [comment]
This is tripping the same assert as #6416

File: lib/std/meta.zig [hunk 1]
```diff
@@ -826,3 +826,75 @@ test "sizeof" {
...
+        .Struct = std.builtin.TypeInfo.Struct{
+            .is_tuple = true,
+            .layout = .Auto,
+            .decls = &[_]std.builtin.TypeInfo.Declaration{},
+            .fields = &argument_field_list,
+        },
+    });
+}
+
+comptime {
```
Comment (alexnask) [inline lib/std/meta.zig:870]: Please move these tests in a test block, at the moment these will run every time meta is referenced.

**Review by alexnask — CHANGES_REQUESTED** [review]

**Review by alexnask — APPROVED** [review]

**Comment by alexnask** [comment]
Maybe thsi can use std.meta.Tuple instead now?  
Although it would be a bit less efficient.

**Comment by MasterQ32** [comment]
I thought about it too, but i think it's good as it is. we don't have to waste more `comptime` time than necessary

**Pull request merged**